src/pages/home/report/ReportActionsList.js

```diff
@@ -276,31 +277,55 @@ function ReportActionsList({
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [report.reportID]);
 
+    /**
+     * Check if the new floating message counter needs to be shown/hidden.
+     * @returns {boolean}
+     */
+    const showFloatingMessageCounter = () => {
+        const hasUnreadMessageAndFloatingMessageIsHidden = !isFloatingMessageCounterVisible && !!currentUnreadMarker;
+        const isFocusOutsideUnreadMessage = isFloatingMessageCounterVisible && !isUnreadMessageFocused.current;
+
+        if (isFloatingMessageCounterVisible && isUnreadMessageFocused.current) {
+            isUnreadMessageFocused.current = false;
+        }
+
+        return hasUnreadMessageAndFloatingMessageIsHidden || isFocusOutsideUnreadMessage;
+    };
+
     /**
      * Show/hide the new floating message counter when user is scrolling back/forth in the history of messages.
      */
     const handleUnreadFloatingButton = () => {
-        if (scrollingVerticalOffset.current > VERTICAL_OFFSET_THRESHOLD && !isFloatingMessageCounterVisible && !!currentUnreadMarker) {
-            setIsFloatingMessageCounterVisible(true);
-        }
+        const showMessageCounter = showFloatingMessageCounter();
+
+        setIsFloatingMessageCounterVisible(showMessageCounter);
 
         if (scrollingVerticalOffset.current < VERTICAL_OFFSET_THRESHOLD && isFloatingMessageCounterVisible) {
             if (readActionSkipped.current) {
                 readActionSkipped.current = false;
                 Report.readNewestAction(report.reportID);
             }
-            setIsFloatingMessageCounterVisible(false);
         }
     };
 
     const trackVerticalScrolling = (event) => {
         scrollingVerticalOffset.current = event.nativeEvent.contentOffset.y;
-        handleUnreadFloatingButton();
         onScroll(event);
     };
 
-    const scrollToBottomAndMarkReportAsRead = () => {
-        reportScrollManager.scrollToBottom();
+    const getScrollIndex = () => {
+        const bottomIndex = 0;
+        const firstUnreadMessageIndex = _.findIndex(sortedReportActions, (reportAction) => reportAction.reportActionID === currentUnreadMarker);
+
+        return firstUnreadMessageIndex > -1 ? firstUnreadMessageIndex : bottomIndex;
+    };
+
+    const scrollToUnreadMessageAndMarkReportAsRead = () => {
+        const scrollIndex = getScrollIndex();
+
+        isUnreadMessageFocused.current = true;
```

Comment:
This flag is used to avoid showing the "**New Message**" button when the scroll ends after the user clicks the "**New Message**" button. 

When the user clicks on the "**New Message**" button the scroll occurs and we don't need to show again the "**New Message**" button because this current scroll was triggered by the "**New Message**" button action.